niworkflows/viz/plots.py

```diff
@@ -644,9 +644,13 @@ def compcor_variance_plot(metadata_files, metadata_sources=None,
                           linewidth=2, linestyle=':')
             ax[m].text(0, 100 * thr, '{:.0f}'.format(100 * thr),
                        fontsize='x-small', bbox=bbox_txt)
-            ax[m].text(varexp[thr][0], 25,
+            try:
+                n_components = varexp[thr][0]
+            except IndexError:
+                n_components = varexp[thr]
+            ax[m].text(n_components, 25,
                        '{} components explain\n{:.0f}% of variance'.format(
-                            varexp[thr][0], 100 * thr),
+                            n_components, 100 * thr),
```

Comment:
I think the cleaner fix would actually be to change L640-L641:

```Diff
             varexp[thr] = np.searchsorted(
-                components['cumulative_variance_explained'], thr) + 1
+                np.atleast1d(components['cumulative_variance_explained']), thr) + 1
```

Then `varexp[thr]` is always an array, for any of the uses later.